src/services/datasource.ts

```diff
@@ -50,35 +70,151 @@ class WrappedLokiDatasource extends RuntimeDataSource<DataQuery> {
             throw new Error('Invalid datasource!');
           }
 
-          const requestType = request.targets?.[0]?.resource;
+          const dataQueryRequest = { ...request };
+          // override the target datasource to Loki
+          dataQueryRequest.targets = request.targets.map((target) => {
+            target.datasource = ds;
+            return target;
+          });
+
+          dataQueryRequest.targets.forEach((target) => {
+            const requestType = target?.resource;
+
+            switch (requestType) {
+              case 'volume': {
+                this.getVolume(dataQueryRequest, ds, subscriber);
+                break;
+              }
+              case 'patterns': {
+                this.getPatterns(dataQueryRequest, ds, subscriber);
+                break;
+              }
+              default: {
+                this.getData(dataQueryRequest, ds, subscriber);
```

Comment:
After reviewing `getData`, aren't we sending multiple `targets` when we do this in the loop through `dataQueryRequest.targets`? I guess it should happen outside, maybe? Otherwise, not sure I follow.

Reply:
Yeah good catch, I added an assertion that we only have a single resource per request

Reply:
Aaaand removed the unnecessary iteration